src/ucp/tag/rndv.c

```diff
@@ -29,8 +29,9 @@ static int ucp_tag_rndv_is_get_op_possible(ucp_ep_h ep, uct_rkey_t rkey)
 
 static void ucp_rndv_rma_request_send_buffer_dereg(ucp_request_t *sreq)
 {
-    if ((UCP_DT_IS_CONTIG(sreq->send.datatype)) &&
-        (ucp_ep_is_rndv_lane_present(sreq->send.ep))) {
+    if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
+        ucp_ep_is_rndv_lane_present(sreq->send.ep) &&
+        ucp_request_is_send_buffer_reg(sreq) /* TODO: How is this possible? */) {
```

Comment:
@evgeny-leksikov shm is the alias in gtest, not mm :)
@yosefe ucp_rndv_rma_request_send_buffer_dereg can be called from AM flow but in has protections inside so that only what was registered would be deregistered:
```
if (req->send.state.dt.contig.memh != UCT_MEM_HANDLE_NULL) {
     uct_md_mem_dereg(uct_md, state->dt.contig.memh);
}
```